core/src/avm2/globals/xml.rs

```diff
@@ -164,13 +164,25 @@ pub fn parent<'gc>(
 pub fn elements<'gc>(
     activation: &mut Activation<'_, 'gc>,
     this: Option<Object<'gc>>,
-    _args: &[Value<'gc>],
+    args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
     let xml = this.unwrap().as_xml_object().unwrap();
+    let element_name = if args[0] == Value::Undefined {
+        AvmString::from("*")
+    } else {
+        args[0].coerce_to_string(activation)?
+    };
+    let is_asterisk = element_name == AvmString::from("*");
```

Comment:
You can use `name_to_multiname` to simplify your code.

Reply:
Done, thank you for pointing me to the function. I also made name_to_multiname resolve an undefined into a "*"